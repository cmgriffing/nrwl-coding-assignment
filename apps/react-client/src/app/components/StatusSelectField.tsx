import React from 'react';

interface StatusSelectFieldProps {
  onChange: (completed: boolean) => Promise<void> | void;
  completed: boolean;
  assigned: boolean;
}

export function StatusSelectField({
  onChange,
  completed,
  assigned,
}: StatusSelectFieldProps) {
  return (
    <select
      value={completed ? 'true' : undefined}
      onChange={(event) => {
        onChange(event.currentTarget.value === 'true');
      }}
    >
      <option value={undefined}>{assigned ? 'Pending' : 'Unassigned'}</option>
      <option value="true">Completed</option>
    </select>
  );
}
