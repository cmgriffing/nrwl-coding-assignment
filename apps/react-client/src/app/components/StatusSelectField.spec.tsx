import React from 'react';

import { render } from '@testing-library/react';
import '@testing-library/jest-dom';

import { StatusSelectField } from './StatusSelectField';

it('renders a Pending state', () => {
  const { asFragment, getByRole } = render(
    <StatusSelectField completed={false} assigned={false} onChange={() => {}} />
  );

  expect(asFragment).toMatchSnapshot();
  expect(
    (getByRole('option', { name: 'Unassigned' }) as HTMLOptionElement).selected
  ).toEqual(true);
});

it('renders an Assigned state', () => {
  const { asFragment, getByRole } = render(
    <StatusSelectField completed={false} assigned={true} onChange={() => {}} />
  );
  expect(asFragment()).toMatchSnapshot();
  expect(
    (getByRole('option', { name: 'Pending' }) as HTMLOptionElement).selected
  ).toEqual(true);
});

it('renders a Completed state', () => {
  const { asFragment, getByRole } = render(
    <StatusSelectField completed={true} assigned={false} onChange={() => {}} />
  );
  expect(asFragment()).toMatchSnapshot();
  expect(
    (getByRole('option', { name: 'Completed' }) as HTMLOptionElement).selected
  ).toEqual(true);
});
