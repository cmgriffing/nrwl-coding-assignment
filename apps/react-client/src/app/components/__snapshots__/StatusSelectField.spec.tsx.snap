// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a Completed state 1`] = `
<DocumentFragment>
  <select>
    <option>
      Unassigned
    </option>
    <option
      value="true"
    >
      Completed
    </option>
  </select>
</DocumentFragment>
`;

exports[`renders a Pending state 1`] = `[Function]`;

exports[`renders an Assigned state 1`] = `
<DocumentFragment>
  <select>
    <option>
      Pending
    </option>
    <option
      value="true"
    >
      Completed
    </option>
  </select>
</DocumentFragment>
`;
